### PR TITLE
Fixed Verify Breakpoints

### PR DESCRIPTION
**Pull request by Wopque**

### CODE REVIEW

File: src/debug.ts [hunk 1]
```diff
@@ -110,26 +110,33 @@ export class DashmipsDebugSession extends LoggingDebugSession {
     protected async launchRequest(response: DebugProtocol.LaunchResponse, args: LaunchRequestArguments) {
         this.config = args
         this.runInTerminalRequest(...buildTerminalLaunchRequestParams(args))
-        await this.configurationDone.wait(1500)
+        await this.configurationDone.wait(1000)
+        await this.configurationDone.wait(900)
```
Comment (Wopque) [inline src/debug.ts:114]: After testing discovered we need to wait a bit longer to get the program to run without breakpoints set.

File: src/debug.ts [hunk 2]
```diff
@@ -94,6 +93,7 @@ export class DashmipsDebugSession extends LoggingDebugSession {
         response.body.supportsEvaluateForHovers = true
         response.body.supportsStepBack = false
         response.body.supportsValueFormattingOptions = true
+        response.body.supportsTerminateRequest = true
```
Comment (nbbeeken) [inline src/debug.ts:96]: 🚀 🚀 🚀

File: src/debug.ts [hunk 3]
```diff
@@ -110,22 +110,33 @@ export class DashmipsDebugSession extends LoggingDebugSession {
     protected async launchRequest(response: DebugProtocol.LaunchResponse, args: LaunchRequestArguments) {
         this.config = args
         this.runInTerminalRequest(...buildTerminalLaunchRequestParams(args))
+        await this.configurationDone.wait(1000)
+        await this.configurationDone.wait(1000)
```
Comment (nbbeeken) [inline src/debug.ts:114]: Is this duplicated by accident?

File: src/debug.ts [hunk 4]
```diff
@@ -147,20 +158,25 @@ export class DashmipsDebugSession extends LoggingDebugSession {
             } as DashmipsBreakpointInfo
         })
 
-        await this.client.ready()
+        // We need to block here until the socket is open
+        if (!this.client.open) {
+            await this.configurationDone.wait(1000)
+            await this.configurationDone.wait(1100)
+        }
```
Comment (nbbeeken) [inline src/debug.ts:165]: Same as above, two calls?

File: src/debug.ts [hunk 5]
```diff
@@ -147,20 +158,25 @@ export class DashmipsDebugSession extends LoggingDebugSession {
...
+
         this.client.call('verify_breakpoints', this.breakpoints)
-        this.client.once('verify_breakpoints', ([vscodeBreakpoints, _]) => {
+        this.client.once('verify_breakpoints', ([vscodeBreakpoints, locations]) => {
             response.body = {
                 breakpoints: vscodeBreakpoints.map(
                     (bp, idx) =>
                         new Breakpoint(
-                            false,
+                            locations[idx] != -1,
```
Comment (nbbeeken) [inline src/debug.ts:173]: Let's comment why -1 is special

**Comment by nbbeeken** [comment]
Looks like you have merge conflicts, you'll have to resolve those, you can merge master into this branch, resolve, and push to this branch, then the github warnings will go away 😃

File: src/debug.ts (hunk 4 above)
Comment (Wopque) [reply src/debug.ts:166]: After testing, I found that the python code needed a bit longer to run if there were no breakpoints set.  However, adding more time to a single wait call for some reason does not solve the problem (I've tried 2000, 3000, etc...) but adding two separate calls works.  Not sure why this happens, if you know why I'd be happy to find an alternative solution.

Comment (Wopque) [reply src/debug.ts:166]: Also I managed to resolve the conflict, but after adding the code again there is another conflict with the master.  Does this mean that the code cannot be ever added/merged due to conflicts?

**Pull request merged**